### PR TITLE
Do not handle redirects multiple times

### LINKED ISSUE #12: Add redirects to queue
The crawler immediately follows redirects instead of adding the target to the queue.

One disadvantage of this is that the target page is processed again, even if it has already been crawled.

### PR DESCRIPTION
**Pull request by Toflar**
Fixes https://github.com/terminal42/escargot/issues/12

### CODE REVIEW

**Comment by Mynyx** [comment]
I would prefer not to immediately follow redicrets (https://symfony.com/doc/current/components/http_client.html#redirects) and add the target of a redirect to the queue.

For example, it isn't currently checked whether the target of a redirect is prohibited by a robots.txt?

**Comment by Toflar** [comment]
This would mean we need to handle redirects on our own. I don't really feel like building that but PR's welcome :)

**Comment by Mynyx** [comment]
I have now tested the PR. Unfortunetly it isn't working, because ``redirect_url`` is ``null``.

**Comment by Mynyx** [comment]
Maybe ``$response->getInfo('url')`` could be a solution instead. It seems to work.

>     *  - url (string) - the last effective URL of the request

https://github.com/symfony/http-client-contracts/blob/master/ResponseInterface.php#L97

**Comment by Toflar** [comment]
Ah, I've misunderstood the docs then. Good find :)

File: src/Escargot.php [hunk 1]
```diff
@@ -419,8 +419,8 @@ private function processResponseChunk(ResponseInterface $response, ChunkInterfac
                 // If the response was a redirect of an URI we have already crawled, we can early abort
                 // this response as it has already been processed.
                 if ($response->getInfo('redirect_count') > 0
-                    && $response->getInfo('redirect_url')
-                    && null !== $this->queue->get($this->getJobId(), CrawlUri::normalizeUri(new Uri((string) $response->getInfo('redirect_url'))))
+                    && $response->getInfo('url')
```
Comment (Mynyx) [inline src/Escargot.php:422]:
```suggestion
```

``url`` can not be ``null`` according to the ResponseInterface docs.

**Review by Mynyx — CHANGES_REQUESTED** [review]

**Pull request merged**